nkms/storage/encrypted_file.py

```diff
@@ -120,51 +47,28 @@ def _read_chunk(self, chunk_size, nonce):
         :return: Decrypted/Authenticated chunk
         :rtype: Bytes
         """
-        ciphertext = self.file_obj.read(chunk_size)
+        ciphertext = self.file_obj.read(chunk_size + PADDING_LENGTH)
         return self.cipher.decrypt(ciphertext, nonce=nonce)
 
-    def open_new(self, keys, chunk_size=1000000, nonce=None):
-        """
-        Opens a new EncryptedFile and creates a header for it ready for
-        writing encrypted data.
-
-        :param list keys: Encrypted keys to put in the header.
-        :param int chunk_size: Size of encrypted chunks in bytes, default is 1MB
-        :param bytes nonce: 20 byte Nonce to use for encryption
-        """
-        self.file_obj = open(self.path, mode=self.mode)
-        self._build_header(nonce=nonce, keys=keys, chunk_size=chunk_size)
-
-    def open(self, is_new=False):
-        """
-        Opens a file for Encryption/Decryption.
-
-        :param bool is_new: Is the file new (and empty)?
-        """
-        # TODO: Error if self.file_obj is already defined
-        self.file_obj = open(self.path, mode=self.mode)
-
-        # file_obj is now ready for reading/writing encrypted data
-        if not is_new:
-            self._read_header()
-
     def read(self, num_chunks=0):
         """
         Reads num_chunks of encrypted ciphertext and decrypt/authenticate it.
 
-        :param int num_chunks: Number of chunks to read. Default is all chunks
+        :param int num_chunks: Number of chunks to read. When set to 0, it will
```

Comment:
The idea was to use it as a file. Will it work correctly when the parameter is num_chunks, not num_bytes?

Reply:
It already reads by `num_chunks`. Reading per byte is pretty useless because there's nothing that could possibly be done with it besides collecting enough ciphertext to decrypt in the first place.

Reply:
Am I right that EncryptedFile object will be returned to user to deal with it as with a normal file?

Reply:
Oh well, I guess not. We're getting chunks, not stream of bytes..

Reply:
It's a compelling debate though.  But at the end of the day, what are the main things we imagine they'd do with a file-like object that they can't do with `EncryptedFile`?  Pass it to... what exactly?